control/lane_departure_checker/src/lane_departure_checker_node/lane_departure_checker_node.cpp

```diff
@@ -51,41 +52,48 @@ std::array<geometry_msgs::msg::Point, 3> triangle2points(
 }
 
 lanelet::ConstLanelets getRouteLanelets(
-  const lanelet::LaneletMap & lanelet_map, const lanelet::routing::RoutingGraphPtr & routing_graph,
-  const std::vector<HADMapSegment> & route_sections, const double vehicle_length)
+  const lanelet::LaneletMapPtr & lanelet_map,
+  const lanelet::routing::RoutingGraphPtr & routing_graph,
+  const autoware_auto_planning_msgs::msg::HADMapRoute::ConstSharedPtr & route_ptr,
+  const double vehicle_length)
 {
   lanelet::ConstLanelets route_lanelets;
 
+  bool is_route_valid = lanelet::utils::route::isRouteValid(*route_ptr, lanelet_map);
+  if (!is_route_valid) {
+    return route_lanelets;
+  }
+
   // Add preceding lanes of front route_section to prevent detection errors
   {
     const auto extension_length = 2 * vehicle_length;
 
-    for (const auto & primitive : route_sections.front().primitives) {
+    for (const auto & primitive : route_ptr->segments.front().primitives) {
```

Comment:
Did you make sure that size of segment is more than 1 all the time somewhere else?

Reply:
`route_ptr` is the same as subscribed `~/input/route` (`const HADMapRoute::ConstSharedPtr`)
I  think if it is subscribed, `route_ptr` is not null.

Reply:
route_ptr's size is empty then it will die

Reply:
Got it and fixed by https://github.com/autowarefoundation/autoware.universe/pull/1153/commits/0349a6bf4c54fc5e977b85dea92c0f1fe907af2f.